src/Sylius/Component/Locale/Provider/LocaleCollectionProviderInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Locale\Provider;
+
+use Sylius\Component\Locale\Model\LocaleInterface;
+
+interface LocaleCollectionProviderInterface
+{
+    /**
+     * @return array<array-key, LocaleInterface>
+     */
+    public function getAll(): array;
```

Comment:
Wondering whether we shouldn't introduce it more softly, instead of a whole new interface and classes, just use a `@method` note on the existing `LocaleProviderInterface`. We'd still be BC compliant, and we'd have a bit less code to worry about.
Something like that has already been done in `Sylius\Component\Core\Calculator\ProductVariantPricesCalculatorInterface`

Reply:
There is a `ChannelBasedLocaleProvider` class which acts as a decorator for the `LocalProvider` class and implements the `LocalProviderInterfece`,  making it more complicated to make in this way. We have decided to leave this in its current form and address it in another PR.